tf_files/aws/modules/rds/main.tf

```diff
@@ -0,0 +1,168 @@
+
+
+locals {
+  is_mssql = "${element(split("-", var.rds_instance_engine), 0) == "sqlserver" ? true : false}"
+}
+
+data "aws_iam_policy_document" "enhanced_monitoring" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["monitoring.rds.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "enhanced_monitoring" {
+  count = "${var.rds_instance_create_monitoring_role ? 1 : 0}"
+
+  name               = "${var.rds_instance_monitoring_role_name}"
+  assume_role_policy = "${data.aws_iam_policy_document.enhanced_monitoring.json}"
+
+  tags = "${merge(map("Name", format("%s", var.rds_instance_monitoring_role_name)), var.rds_instance_tags )}"
+}
+
+resource "aws_iam_role_policy_attachment" "enhanced_monitoring" {
+  count = "${var.rds_instance_create_monitoring_role ? 1 : 0}"
+
+  role       = "${aws_iam_role.enhanced_monitoring.name}"
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonRDSEnhancedMonitoringRole"
+}
+
+resource "aws_db_instance" "this" {
+  count = "${var.rds_instance_create && false == local.is_mssql ? 1 : 0}"
+
+  identifier                            = "${var.rds_instance_identifier}"
```

Comment:
If it's a postgres database - can we do the trick to crank up the limit on the number of connections?

Reply:
that kind of configurations are managed by "Parameter groups" which is a different resource. We could create a new module that would create that kind of resource and use the output as input for this one. 

Adding it in here would defeat the purpose of simplistic modules.